src/test/java/org/gridsuite/modification/modifications/GeneratorScalingTest.java

```diff
@@ -217,6 +217,54 @@ void testScalingCreationWithWarning() throws Exception {
         assertEquals(300, getNetwork().getGenerator(GENERATOR_ID_10).getTargetP(), 0.01D);
     }
 
+    @Test
+    void testScalingCreationWithDupeGenerator() throws Exception {
+        // GENERATOR 10 is in both filter
+        // the filter with wrong ids will be taken for valid ids
+        // it will be counted twice
+        // as his target p is 100MW and target p of GENERATOR 9 is 200MW
+        // both will be up of 300MW
+        Map<UUID, FilterEquipments> filters = Map.of(FILTER_ID_5, FilterEquipments.builder()
+                .filterId(FILTER_ID_5)
+                .identifiableAttributes(List.of(new IdentifiableAttributes(GENERATOR_ID_9, IdentifiableType.GENERATOR, 0.0),
+                    new IdentifiableAttributes(GENERATOR_ID_10, IdentifiableType.LOAD, 9.0)))
+                .build(),
+            FILTER_WRONG_ID_2, FilterEquipments.builder()
+                .filterId(FILTER_WRONG_ID_2)
+                .identifiableAttributes(
+                    List.of(new IdentifiableAttributes(GENERATOR_WRONG_ID_1, IdentifiableType.GENERATOR, 2.0),
+                        new IdentifiableAttributes(GENERATOR_ID_10, IdentifiableType.LOAD, 9.0)))
+                .build());
+        when(filterService.getUuidFilterEquipmentsMap(any(), any())).thenReturn(filters);
+
+        var filter = FilterInfos.builder()
+            .name("filter")
+            .id(FILTER_WRONG_ID_2)
+            .build();
+
+        var filter2 = FilterInfos.builder()
+            .name("filter2")
+            .id(FILTER_ID_5)
+            .build();
+
+        var variation = ScalingVariationInfos.builder()
+            .variationMode(VariationMode.PROPORTIONAL)
+            .variationValue(900D)
+            .filters(List.of(filter, filter2))
+            .build();
+        var generatorScalingInfo = GeneratorScalingInfos.builder()
+            .stashed(false)
+            .variationType(VariationType.TARGET_P)
+            .variations(List.of(variation))
+            .build();
+
+        GeneratorScaling generatorScaling = (GeneratorScaling) generatorScalingInfo.toModification();
+        generatorScaling.initApplicationContext(filterService);
+        generatorScaling.apply(getNetwork());
+        assertEquals(500, getNetwork().getGenerator(GENERATOR_ID_9).getTargetP(), 0.01D);
```

Comment:
Here, we target 900 MW at the end, and we have 300 MW before doing the scaling, so we want to add 600 MW ...

As GENERATOR_ID_9 has an initial targetP = 200, and GENERATOR_ID_10 has an initial targetP = 100,
GENERATOR_ID_9 targetP should increase by 400 MW and GENERATOR_ID_10 should increase by 200 MW,

So, at the end, GENERATOR_ID_9 targetP should be 600 MW and GENERATOR_ID_10 targetP should be 300 MW

See my remark above about the sum computation ...

Reply:
no if GENERATOR_ID_10 is twice in the filters it will be counted as twice
its target p is 100MW counted as twice it will be 200 MW 
GENERATOR 9 has a target of 200 MW 
so to get to 900 MW it will be 400 MW for generator_9 500 MW (200 + 300) and 400 MW (100 + 300) for generator 10
as it is there will be the same raise for both generator (as their target p is counted as similar)